pkg/operation/botanist/component/etcdcopybackupstask/etcdcopybackupstask.go

```diff
@@ -0,0 +1,202 @@
+	TargetStore druidv1alpha1.StoreSpec
+	// MaxBackups is the maximum number of backups that will be copied starting with the most recent ones.
+	MaxBackups *uint32
+	// MaxBackupAge is the maximum age in days that a backup must have in order to be copied.
+	MaxBackupAge *uint32
+}
+
+type etcdCopyBackupsTask struct {
+	values              *Values
+	logger              logrus.FieldLogger
+	client              client.Client
+	waitInterval        time.Duration
+	waitSevereThreshold time.Duration
+	waitTimeout         time.Duration
+
+	task *druidv1alpha1.EtcdCopyBackupsTask
+}
+
+// New creates a new instance of Interface
+func New(
+	logger logrus.FieldLogger,
+	client client.Client,
+	values *Values,
+	waitInterval time.Duration,
+	waitSevereThreshold time.Duration,
+	waitTimeout time.Duration) Interface {
+
+	return &etcdCopyBackupsTask{
+		values,
+		logger,
+		client,
+		waitInterval,
+		waitSevereThreshold,
+		waitTimeout,
+		&druidv1alpha1.EtcdCopyBackupsTask{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      values.Name,
+				Namespace: values.Namespace,
+			},
+		},
+	}
+}
+
+// Deploy creates the EtcdCopyBackupsTask resource. If it already exists, it is deleted and then recreated.
+func (e *etcdCopyBackupsTask) Deploy(ctx context.Context) error {
+	e.task.Spec.MaxBackupAge = e.values.MaxBackupAge
+	e.task.Spec.MaxBackups = e.values.MaxBackups
+	e.task.Spec.SourceStore = e.values.SourceStore
+	e.task.Spec.TargetStore = e.values.TargetStore
+	return e.client.Create(ctx, e.task)
+}
+
+// Wait waits until the EtcdCopyBackupsTask is ready
+func (e *etcdCopyBackupsTask) Wait(ctx context.Context) error {
+	if err := extensions.WaitUntilObjectReadyWithHealthFunction(
+		ctx,
+		e.client,
+		e.logger,
+		WaitForConditions,
+		e.task,
+		"EtcdCopyBackupsTask",
+		e.waitInterval,
+		e.waitSevereThreshold,
+		e.waitTimeout,
+		e.checkConditions,
+	); err != nil {
+		e.logger.Error(err)
+		return err
+	}
+	return nil
+}
+
+// Destroy deletes the EtcdCopyBackupsTask resource.
+func (e *etcdCopyBackupsTask) Destroy(ctx context.Context) error {
+	return kutil.DeleteObject(
+		ctx,
+		e.client,
+		e.task,
+	)
+}
+
+// WaitCleanup waits until the EtcdCopyBackupsTask is deleted.
+func (e *etcdCopyBackupsTask) WaitCleanup(ctx context.Context) error {
+	return kutil.WaitUntilResourceDeleted(ctx, e.client, e.task, DefaultInterval)
+}
+
+// SetSourceStore sets the specifications for the object store provider from which backups will be copied
+func (e *etcdCopyBackupsTask) SetSourceStore(store druidv1alpha1.StoreSpec) {
+	e.values.SourceStore = store
+}
+
+// SetTargetStore sets the specifications for the object store provider to which backups will be copied
+func (e *etcdCopyBackupsTask) SetTargetStore(store druidv1alpha1.StoreSpec) {
+	e.values.TargetStore = store
+}
+
+// WaitForConditions waits until the EtcdCopyBackupsTask conditions have been populated by the etcd druid
+func WaitForConditions(obj client.Object) error {
```

Comment:
I think this function should rather not be exported, since it's used only once in this file. The corresponding tests could be deleted as well, since they are mostly duplicate with the `#Wait` tests.
```suggestion
func waitForConditions(obj client.Object) error {
```